app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -53,7 +61,11 @@ def delete
       PushNotification::HttpPush.deliver(event)
       flash[:success] = t('notices.backup_codes_deleted')
       revoke_remember_device(current_user)
-      redirect_to account_two_factor_authentication_path
+      if in_multi_mfa_selection_flow?
+        redirect_to authentication_methods_setup_path
+      else
+        redirect_to account_two_factor_authentication_path
+      end
```

Comment:
@aduth I wanted to call this out. Because Delete is being dual-purposed between simply deleting from the account page but also when cancelling during MFA setup I have added this if-statement so they go to the right place. This came out of removing the extra "choose another..." link and making it work like the rest of the MFA.

Reply:
That makes sense, though I'm curious why this wasn't an issue previously when it was "Cancel" ?

Reply:
It's a good question about the coverage. I did bump into a test which caused this to become a problem. /spec/features/two_factor_authentication/multiple_mfa_sign_up_spec.rb#L13
ended up taking the user to their account page instead of to the add mfa page. I think the test passed before because the other link that isn't there anymore would take you to the add mfa page.